hack/update-mockgen.sh

```diff
@@ -21,17 +21,12 @@ BIN="$(dirname "$(realpath "${BASH_SOURCE[0]}")")/../bin"
 # Source-based mocking for internal interfaces
 "${BIN}/mockgen" -package cloud -destination=./pkg/cloud/mock_cloud.go -source pkg/cloud/interface.go
 "${BIN}/mockgen" -package metadata -destination=./pkg/cloud/metadata/mock_metadata.go -source pkg/cloud/metadata/interface.go
```

Comment:
Is there a reason we can't use a `mocks` folder for cloud/metadata mocks? I think we should be consistent and do it everywhere or nowhere

Reply:
Tried this but was unable to do so due to a circular dependency. I think having a `mocks` directory for the driver package is an organizational improvement but inclined to remove it if you feel strongly about the request.

Reply:
I feel strongly that we should be consistent, I don't really care which way it is.

Reply:
Removed mocks directory